### PR TITLE
Arjun | 1204378118194874 | Ability to view awaiting appointment List

### PR DESCRIPTION
**Pull request by Arjun-Go**
Asana card -> https://app.asana.com/0/1204322126657135/1204378118194874

# Summary

This PR has UI changes to add a new awaiting appointments tab in the appointments module.
Some of the changes include:-
- New tab for awaiting appointments same as that of appointments list view
- Disable view of date picker and today tab in awaiting appointments view
- Show/hide certain columns in awaiting appointments tab. 
   - Show columns like Appointment creation date, Appointment Category, Status
   - Hide columns like Walk-in, Appointment date, and Appointment start and end time.
- Ability to add a new awaiting appointment.

[PR for backend changes](https://github.com/Bahmni/openmrs-module-appointments/pull/109) to show appointment creation date in awaiting appointments tab

# Screenshot
image

### CODE REVIEW

File: src/controllers/manage/list/appointmentsListViewController.js [hunk 1]
```diff
@@ -72,6 +99,23 @@ angular.module('bahmni.appointments')
                 });
             };
 
+            var modifyAppointmentPriorities = function(){
+                $scope.filteredAppointments.map((appointment) => {
+                    var priorityModified = false;
+                    appConfig.config.priorityOptionsList.map((priority) => {
```
Comment (binduak) [inline src/controllers/manage/list/appointmentsListViewController.js:105]: @Arjun-Go what if we don't configure appConfig.config.priorityOptionsList ?

Comment (Arjun-Go) [reply src/controllers/manage/list/appointmentsListViewController.js:105]: @binduak In a scenario where `priorityOptionList` is not present/configured, we would face a broken UI with an undefined or null value error. Now for CURE, we have it configured already [here](https://github.com/cureinternational/cure-bahmni-emr/blob/7324ac3a08e1ea46956284b0ef52cbe66a13769d/openmrs/apps/appointments/app.json#L24). 
But when it comes to Bahmni, we would have a broken UI, I agree.

Comment (gsluthra) [reply src/controllers/manage/list/appointmentsListViewController.js:105]: The code needs to be backward compatible, assuming that people are upgrading to new docker images, but using their older "config" folders.

**Comment by gsluthra** [comment]
Can you please also link to appropriate JIRA card, so folks in the community understand the scope of this PR / commits?

**Comment by Arjun-Go** [comment]
> Can you please also link to appropriate JIRA card, so folks in the community understand the scope of this PR / commits?

We don't have a JIRA card for this issue. I have linked the respective card from Asana board.

File: src/controllers/manage/list/appointmentsListViewController.js (hunk 1 above)
Comment (Arjun-Go) [reply src/controllers/manage/list/appointmentsListViewController.js:105]: Have raised a thread in bahmni-ipd channel for the same - https://bahmni.slack.com/archives/C056DPYNDN3/p1684755171955239

Comment (binduak) [reply src/controllers/manage/list/appointmentsListViewController.js:105]: @gsluthra we need this PR for internal showcase. We will continue the discussion in the #bahmni-ipd channel and will work on as per the inputs from product
on creating JIRA cards for the product code changes, we would like to it take as part of tomorrows architecture call to come up with the process. Hope that is fine.

**Review by binduak — APPROVED** [review]

**Review by binduak — APPROVED** [review]

**Pull request merged**